Exiled.Events/Patches/Events/Player/Shoot.cs

```diff
@@ -29,276 +37,101 @@ namespace Exiled.Events.Patches.Events.Player
-                    __instance.RpcConfirmShot(true, __instance.curWeapon);
-                    __instance.PlaceDecal(true, new Ray(__instance.camera.position, dir), (int)referenceHub.characterClassManager.CurClass, num3);
-                    return false;
-                }
-                else
-                {
-                    if (target != null && hitboxType == HitBoxType.WINDOW && target.GetComponent<BreakableWindow>() != null)
-                    {
-                        float time = Vector3.Distance(__instance.camera.transform.position, target.transform.position);
-                        float damage = __instance.weapons[__instance.curWeapon].damageOverDistance.Evaluate(time);
-                        target.GetComponent<BreakableWindow>().ServerDamageWindow(damage);
-                        __instance.RpcConfirmShot(true, __instance.curWeapon);
-                        return false;
-                    }
-
-                    __instance.PlaceDecal(false, new Ray(__instance.camera.position, dir), __instance.curWeapon, 0f);
-                    __instance.RpcConfirmShot(false, __instance.curWeapon);
-                    return false;
-                }
-            }
-            catch (Exception e)
+                // Player.Get(this.gameObject)
+                new CodeInstruction(OpCodes.Ldarg_0).MoveLabelsFrom(newInstructions[index]),
+                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(WeaponManager), nameof(WeaponManager.gameObject))),
+                new CodeInstruction(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(GameObject) })),
+
+                // GameObject target
+                new CodeInstruction(OpCodes.Ldarg_1),
+
+                // Vector3 targetPos
+                new CodeInstruction(OpCodes.Ldarg_S, 5),
+
+                // true
+                new CodeInstruction(OpCodes.Ldc_I4_1),
+
+                // var ev = new ShootingEventArgs(...)
+                new CodeInstruction(OpCodes.Newobj, GetDeclaredConstructors(typeof(ShootingEventArgs))[0]),
+                new CodeInstruction(OpCodes.Dup),
+                new CodeInstruction(OpCodes.Dup),
+                new CodeInstruction(OpCodes.Stloc_S, ev1.LocalIndex),
+
+                // Handlers.Server.OnShooting(ev)
+                new CodeInstruction(OpCodes.Call, Method(typeof(Handlers.Player), nameof(Handlers.Player.OnShooting))),
+
+                // if (!ev.IsAllowed)
+                //    return;
+                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(ShootingEventArgs), nameof(ShootingEventArgs.IsAllowed))),
+                new CodeInstruction(OpCodes.Brfalse, returnLabel),
+            });
+
+            // Search for the only "PlayerStats.HurtPlayer()"
+            offset = 34;
```

Comment:
This is fine, but is this the nearest unique opcode?

Reply:
ldstr "W.11" or even div would technically be closer, but in the IL viewer they are above the IL call I want to insert above, but in the C# code they are below it, in a switch-case, and I wasn't certain how that works. If I got the index of the last `div` call, and offset it by +2, in the IL code it would be right where I want it, but that IL call is in a switch case that's only used in switch-case, so i wasn't sure how it would behave.

Reply:
Yeah, that's normal, you can safely use that `div` call.